lib/services/elasticsearch.js

```diff
@@ -222,41 +222,37 @@ class ElasticSearch extends Service {
       scroll
     };
 
-    debug('Search: %o', esRequest);
+    debug('Search: %j', esRequest);
 
-    return this._client.search(esRequest)
-      .then(({ body }) => {
-        if (body._scroll_id) {
-          const
-            // ms(scroll) may return undefined if in microseconds or in nanoseconds
-            ttl = esRequest.scroll && ms(esRequest.scroll)
-              || ms(this._config.defaults.scrollTTL),
-            key = scrollCachePrefix + this._kuzzle.constructor.hash(
-              body._scroll_id
-            );
-
-          return this._kuzzle.cacheEngine.internal.psetex(key, ttl, 0)
-            .then(() => this._formatSearchResult(body));
-        }
+    let body;
+    try {
+      ({ body } = await this._client.search(esRequest));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment of "{ body }" from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AAssignmentWithinCondition)